Refactor email reminders to remove non-magic link

Cruikshanks opened the pull request:
The original plan was to ship a version of the first reminder email that only asked users to re-register. This was because it was thought shipping renewals would take some time.

Thanks to the success of the team, we actually have renewals including the version of the email which includes a **magic link** ready to go. This means the old re-register email will never see the light of day.

However we still have code that makes distinctions between the re-register and magic link versions of the first email reminder. So this change is some housekeeping to clean up the logic around the first email reminder to remove those distinctions.

<details>
<summary>First reminder</summary>
<img width="423" alt="Screenshot 2019-08-13 at 17 10 07" src="https://user-images.githubusercontent.com/1789650/62958272-46405c80-bdee-11e9-8d78-e89e5a95296d.png">

</details>

<details>
<summary>Second reminder</summary>
<img width="399" alt="Screenshot 2019-08-13 at 17 11 13" src="https://user-images.githubusercontent.com/1789650/62958127-fe213a00-bded-11e9-9ce2-b29bfc3e0b18.png">

</details>

Cruikshanks committed:
Refactor email reminders to remove non-magic link

The original plan was to ship a version of the first reminder email that only asked users to re-register. This was because it was thought shipping renewals would take some time.

Thanks to the success of the team, we actually have renewals including the version of the email which includes a **magic link** ready to go. This means the old re-register email will never see the light of day.

However we still have code that makes distinctions between the re-register and magic link versions of the first email reminder. So this change is some housekeeping to clean up the logic around the first email reminder to remove those distinctions.

Cruikshanks committed:
Remove references to alternate 'magic link' email

This change either removes anything that refers to an alternate magic link email eg the test rake task and some tests, or replaces the content of the first reminder with stuff from the magic link version.

Cruikshanks committed:
Fix incorrect title in 2nd reminder test

We say 4 weeks in the description when actually we are testing for 2 weeks.

Cruikshanks committed:
Make reminder feature toggles consistent

We have feature toggles for both the first and second reminder emails. However only the second one could be overridden with an env var. This change just makes things consistent.

Cruikshanks committed:
Add tests for second reminder mail

Whilst going through the code for this refactor, spotted that we had test coverage for the first email reminder, and at the time for the magic link reminder. But we didn't have coverage for the second.

This change just makes things consistent by adding some tests for second email reminder mailer.

Cruikshanks commented:
Sorry, jumped the gun a bit. Should check the emails are actually ok before asking for a review!

cintamani commented on app/mailers/renewal_reminder_mailer.rb:
```diff
@@ -7,11 +7,12 @@ class RenewalReminderMailer < ActionMailer::Base
 
   def first_reminder_email(registration)
     assign_values_for_email(registration)
+    @magic_link_url = magic_link_url(registration)
```
Can we move this inside the `assign_values_for_email` method?

cintamani commented on spec/mailers/renewal_mailer_spec.rb:
```diff
@@ -72,11 +73,16 @@
   end
 
-  describe "first_renew_with_magic_link_email" do
+  describe "second_reminder_email" do
```
I missed this! Thanks!

Cruikshanks committed:
Move magic_link to assign_values_for_email

Now we are not making a distinction about whether to include the magic link, we can assign it in the `assign_values_for_email()` method.

Cruikshanks commented:
Ok, checked the emails now. As far as I can see I haven't mucked anything up, so this is good to go for a review (thanks for the initial heads up @cintamani 😁 )

cintamani reviewed (approved):
:)
